packages/core/src/StateNode.ts

```diff
@@ -134,16 +134,18 @@ const validateArrayifiedTransitions = <TContext>(
     .slice(0, -1)
     .some(
       transition =>
-        isString(transition.target) ||
-        isMachine(transition.target) ||
-        (!('cond' in transition) && !('in' in transition))
+        !('cond' in transition) &&
+        !('in' in transition) &&
+        (isString(transition.target) || isMachine(transition.target))
     );
   const eventText =
     event === NULL_EVENT ? 'the transient event' : `event '${event}'`;
 
   warn(
     !hasNonLastUnguardedTarget,
-    `One or more transitions for ${eventText} on state '${stateNode.id}' are unreachable. ` +
+    `One or more transitions for ${eventText} on state '${
```

Comment:
this reformatting is super annoying, tried to look briefly into fixing this - but couldn't locate the problem yet :s

Reply:
It's alright, we'll just deal with it 😅